### PR TITLE
Add `HTTP::Request#form_params`

### PR DESCRIPTION
**Pull request by threez**
The current API has helpers to get the query params for an HTTP request. I was searching io how to handle 
a regular http post. I did not find an direct answer, again searching the docs. After a little while I noticed that there are outdated examples on stack overflow. To safe everybody else the hassle of searching how to get form params, I added
convenience helpers in the spirit of `query_params`. 

This will enable uses like this:

```crystal
require "http/server"

class SomeHandler
  include HTTP::Handler

  def call(context)
    if params = context.request.form_params?
      # do something with
      puts params["some"]?
    end
  end
end


```

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

File: src/http/request.cr [hunk 1]
```diff
@@ -80,6 +81,24 @@ class HTTP::Request
...
 
+  # Returns a convenience wrapper to parse form params, see `URI::Params`.
+  # Returns `nil` in case the content type `"application/x-www-form-urlencoded"`
+  # is not present or the body is `nil`.
+  def form_params? : HTTP::Params?
+    @form_params ||= begin
+      if headers["Content-Type"]? == "application/x-www-form-urlencoded"
+        if _body = body
+          HTTP::Params.parse(_body.gets_to_end)
+        end
```
Comment (HertzDevil) [inline src/http/request.cr:92]: Variables starting with an underscore are often used for unused variables, and some linters will pick this up. Simply use a method call on `self` here:
```suggestion
        if body = self.body
          HTTP::Params.parse(body.gets_to_end)
        end
```

**Review by HertzDevil — CHANGES_REQUESTED** [review]

**Comment by straight-shoota** [comment]
@threez Would you mind adressing the review comment?
If you don't want to continue with this PR, please let us know and someone else will take over.

File: src/http/request.cr (hunk 1 above)
Comment (threez) [reply src/http/request.cr:92]: Thanks for your review.

**Comment by threez** [comment]
@straight-shoota, sorry for late reply. I added the changes as requested.

**Review by HertzDevil — APPROVED** [review]

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**